### ISSUE TITLE
[Feature Request]: Ability to play audio files & voice messages directly within SimpleX

### ISSUE THREAD

**Issue by RetroAbstract** [simplex-chat/simplex-chat#1445]
Currently you have to download a received audio file to listen to it.

This feature would allow the user to listen to received audio files & voice messages directly within the SimpleX app, including audio files & voice messages they've sent.

This would be identical to how audio files & voice messages are handled in Signal & Session for example.

Exciting to see how fast SimpleX has improved and improving, with the desktop client being what will make me make it my daily driver for messaging. Keep up the great work!

**Comment by epoberezkin** [comment]
thank you!

That’s already working in beta versions (and you can only send voice messages from beta too)! The full release is planned for December 6, you can install beta from TestFlight or from Play store beta channel. You’d only be able to switch back to the main version after the full release is done though.

**Comment by epoberezkin** [comment]
Actually, installing 4.3 should make existing messages also playable in the UI, they are not just files.

**Comment by RetroAbstract** [comment]
> Actually, installing 4.3 should make existing messages also playable in the UI, they are not just files.

Tested 4.3 Beta.1 with a friend and only recorded voice messages play in the UI directly. Audio files still don't unfortunately.

**Comment by epoberezkin** [comment]
Yes, it was a conscious design choice. If we do audio files support they should look something different I think.

**Comment by RetroAbstract** [comment]
> Yes, it was a conscious design choice. If we do audio files support they should look something different I think.

I see. 

Audio files support, as in being able to play audio files directly within the UI is a feature my friend and I would really like to see, we couldn't make SimpleX our main messaging app without it frankly. Any chance that could happen?

**Comment by lorenzolamasse** [comment]
I do agree, on this rather old request, all messenger apps i've tested have this ability, not simplex, it's alas a showstopper for me if I can't listen to audio files in app... Not sure how complicated that is, must be platform dependent.
